src/image-upload/image.service.ts

```diff
@@ -0,0 +1,35 @@
+import { Injectable } from '@angular/core';
+import { Headers, Http, RequestOptions, RequestOptionsArgs, Response } from '@angular/http';
+import { Observable } from 'rxjs/Observable';
+
+@Injectable()
+export class ImageService {
+
+  constructor(private http: Http) { }
+
+  public postImage(url: string, image: File, headers?: any[], partName: string = 'image',
+    withCredentials?: boolean): Observable<Response> {
+
+    if (!url || url === '') {
+      throw new Error('Url is not set! Please set it before doing queries');
+    }
+
+    let options: RequestOptionsArgs = new RequestOptions();
+    if (withCredentials) {
+      options.withCredentials = withCredentials;
+    }
+
+    if (headers) {
+      options.headers = new Headers();
+      for (let header of headers) {
+        let key = Object.keys(header)[0];
+        options.headers.append(key, header[key]);
+      }
+    }
```

Comment:
Yeah, this `options.headers = headers` is not working since `Headers` class has a few methods in it so continue with `options.headers = new Headers()` but again need to set header in expected format otherwise it goes like this `0:[object Object]`.